lib/Server.js

```diff
@@ -785,7 +789,12 @@ class Server {
       // set this so that status helper can identify how the project is being run correctly
       this.options.socket = socket;
 
-      startUnixSocket(this.listeningApp, socket, fullCallback, userCallback);
+      startUnixSocket(
+        this.listeningApp,
+        socket,
+        setupAndListenCallback,
+        userCallback
+      );
```

Comment:
Let rewrite this in style (pseudo code):
```
startUnixSocket(
  this.listeningApp,
  socket,
  (error) => { 
    // Here logic for callback
  }
);
```

Idea is not pass a lot of argument in util, just use closure for this

Reply:
@evilebottnawi I was passing in 2 callbacks because I didn't want `options.onListening` to be called if an error occurred before the server starts listening. But maybe that is not important. If you think it is important, maybe we could have:

```
startUnixSocket(
  this.listeningApp,
  socket,
  (error, startedListening) => { 
    if (startedListening) {
      this.options.onListening(this);
    }
    // other functions that always get called, regardless of if server started listening
  }
);
```